python/lsst/ts/attcpip/at_simulator.py

```diff
@@ -0,0 +1,246 @@
+            name="CmdEvtServer",
+        )
+        # TODO DM-38912 Send "port" via configuration.
+        self.telemetry_server = AtServerSimulator(
+            host=tcpip.LOCALHOST_IPV4,
+            port=6000,
+            log=self.log,
+            dispatch_callback=self.telemetry_dispatch_callback,
+            connect_callback=self.telemetry_connect_callback,
+            name="TelemetryServer",
+        )
+
+        # Keep track of the sequence_id as commands come dripping in. The
+        # sequence ID should raise monotonally without gaps. If a gap is seen,
+        # a NOACK should be returned.
+        self.last_sequence_id: int = 0
+
+        # Dict of command: function.
+        self.dispatch_dict: dict[str, typing.Callable] = {}
+
+        self.load_schemas()
+
+    @abc.abstractmethod
+    def load_schemas(self) -> None:
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    async def cmd_evt_connect_callback(self, server: tcpip.OneClientServer) -> None:
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    async def cmd_evt_dispatch_callback(self, data: typing.Any) -> None:
+        raise NotImplementedError
+
+    async def telemetry_connect_callback(self, server: tcpip.OneClientServer) -> None:
+        """Callback function for when a telemetry client connects or
+        disconnects."""
+        pass
+
+    async def telemetry_dispatch_callback(self, data: typing.Any) -> None:
+        """Asynchronous function to call when data are read and dispatched.
+
+        The received data ignored since the telemetry server is only supposed
+        to send data and not to receive any.
+
+        Parameters
+        ----------
+        data : `dict`[`str`, `typing.Any`]
+            The data sent to the server.
+        """
+        pass
+
+    async def verify_data(self, data: dict[str, typing.Any]) -> bool:
+        """Verify the format and values of the data.
+
+        The format of the data is described at
+        https://github.com/lsst-ts/ts_labview_tcp_json
+        as well as in the JSON schemas in the schemas directory.
+
+        Parameters
+        ----------
+        data : `dict` of `any`
+            The dict to be verified.
+
+        Returns
+        -------
+        bool:
+            Whether the data follows the correct format and has the correct
+            contents or not.
+        """
+        if (
+            CommonCommandArgument.ID not in data
+            or CommonCommandArgument.SEQUENCE_ID not in data
+        ):
+            self.log.error(f"Received invalid {data=}. Ignoring.")
+            return False
+        payload_id = data[CommonCommandArgument.ID].replace("cmd_", "command_")
+        if payload_id not in registry:
+            self.log.error(f"Unknown command in {data=}.")
+            return False
+
+        sequence_id = data[CommonCommandArgument.SEQUENCE_ID]
+        if sequence_id - self.last_sequence_id != 1:
+            return False
```

Comment:
The current implementation in LabVIEW TCP/IP server will not reject any sequence ID from command. It just sends the `NoAck` if the received sequence IDs are not continuous as a notification.

Reply:
Thanks for clarifying that again. That, indeed, is how this function is used.